packages/create-react-app/createReactApp.js

```diff
@@ -613,7 +614,13 @@ function checkIfOnline(useYarn) {
   }
 
   return new Promise(resolve => {
-    dns.lookup('registry.yarnpkg.com', err => {
+    let host = 'registry.yarnpkg.com';
+    // If a proxy is defined, we likely can't resolve external hostnames.
+    // Try to resolve the proxy name as an indication of a connection.
+    if (process.env.https_proxy) {
+      host = url.parse(process.env.https_proxy).hostname;
+    }
+    dns.lookup(host, err => {
       resolve(err === null);
```

Comment:
While we're here, this strict equality makes me uncomfortable -- can we do `== null`?